src/components/menu/menu-popover.tsx

```diff
@@ -1,29 +1,70 @@
-import { type FC, type MutableRefObject, useRef } from 'react'
+import { type FC, type MutableRefObject, useLayoutEffect, useRef, useState } from 'react'
 import { useClickOutside } from '../../hooks/use-click-outside'
 import { useMenuContext } from './menu-context'
 import { ElMenuPopover } from './styles'
 
-export const MenuPopover: FC = ({ children }) => {
+export const calculatePopoverPosition = (
+  container: HTMLElement,
+  popover: HTMLDivElement,
+  setPopoverStyle: React.Dispatch<React.SetStateAction<React.CSSProperties>>,
+  additionalGap = 0,
+) => {
+  const triggerBtn = container.querySelector('[role="button"]')
+  if (triggerBtn) {
+    const viewportHeight = window.innerHeight
+    const popoverHeight = popover.getBoundingClientRect().height
+    const { bottom: triggerBottomPos, height: triggerheight } = triggerBtn.getBoundingClientRect()
+    const gap = 3 // the shadow width or .etc
+    const spaceBelowButton = viewportHeight - triggerBottomPos
+
+    const top = popoverHeight > spaceBelowButton ? 0 - popoverHeight - gap : triggerheight + gap
+
+    setPopoverStyle({ top: top + additionalGap })
+  }
+}
+
+export const MenuPopover: FC<{
+  /**
+   * Optional parameter to adjust the vertical position of the popover
+   *
+   * @default 0
+   */
+  additionalGap?: number
+}> = ({ children, additionalGap }) => {
   const { isOpen, closeMenu, getPopoverProps } = useMenuContext()
   const popoverRef = useRef<HTMLDivElement>(null)
 
+  const [popoverStyle, setPopoverStyle] = useState<React.CSSProperties>({
+    position: 'absolute',
+  })
+
   useClickOutside(popoverRef, closeMenu)
 
-  if (isOpen) {
-    const onClick = (event) => {
-      // close menu if clicked target has parent with below selector
-      const menuItem = event.target.closest('[data-close-menu="true"]')
-      if (menuItem) closeMenu()
+  useLayoutEffect(() => {
+    const container = popoverRef.current?.parentElement
+    if (container && isOpen && popoverRef.current) {
+      calculatePopoverPosition(container, popoverRef.current, setPopoverStyle, additionalGap)
```

Comment:
**question:** Does this work when scrolling the viewport? i.e. if a menu is open and the viewport is scrolled, will this behaviour cause the menu to change position?

**note:** I'm not suggesting we _need_ this behaviour, just wanting to understand what has/has not be explored with the current implementation.

Reply:
yes, I think I need to update the example, thanks for noticing this

![menu auto anchor scroll](https://github.com/user-attachments/assets/8aed38b4-47d8-40a2-a3ca-e2c14c893bed)

Reply:
@nurm717123 Sorry, can you scroll without clicking the mouse? Clicking the mouse obviously causes the menu to close, so then reopening would naturally result in the correct positioning. I'm interested in seeing whether the menu's position will update automatically when it's open and the viewport is scrolled. I suspect it won't reposition, which I think is probably fine for now, so long as we make that clear to consumers via the Storybook docs.

In future, I think we'd want to switch to using the new [CSS anchor positioning](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_anchor_positioning) features; they just don't have enough support right now to make it worthwhile.

Reply:
yes, it need to be reopened

> In future, I think we'd want to switch to using the new [CSS anchor positioning](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_anchor_positioning) features; they just don't have enough support right now to make it worthwhile.

I will look into this 👍 

![menu auto anchor scroll](https://github.com/user-attachments/assets/128f5e3a-eaea-4e33-9f5b-248eccfe02ea)

Reply:
@nurm717123 sorry, that gif looks like you're still clicking and dragging the scroll bar. I'd like to see how the menu behaves when you scroll without any click and drag (i.e. scrolling via a touchpad or mouse wheel).

Reply:
I've put the wrong gif previously, I've updated it with newest, but essentially it shows that currenltly it need to be reopened as you said

Reply:
haha no worries. Thanks for confirming 🙏